### ISSUE TITLE
Bug: `no-unused-vars`: `removeVar` suggestion for FunctionDeclaration produces invalid syntax due to ASI hazard

### ISSUE THREAD

**Issue by Kuldeep2822k** [eslint/eslint#20931]
### Environment

Node version: v24.15.0
npm version:  11.12.1
Local ESLint version: v10.4.0
Global ESLint version: 
Operating System: Windows 11 (NT 10.0.26200)

### What parser are you using?

Default (Espree)

### What did you do?

I ran the `no-unused-vars` rule on some code where I had an unused function declaration. When I used my editor's quick-fix to apply the `removeVar` suggestion on that function,it produced syntactically invalid JavaScript because it didn't account for ASI (Automatic Semicolon Insertion). 

Here's the minimal setup to see it happen:

<details>
<summary>Configuration</summary>

```js
{
  rules: { "no-unused-vars": "error" }
}

```

</details>

```js
const x = 1
function unused() {}
(a) => {}
```

### What did you expect to happen?

The `removeVar` suggestion should either produce syntactically valid JavaScript, or not be offered when removing the declaration would create an ASI hazard.

### What actually happened?

Applying the `removeVar` suggestion produced syntactically invalid JavaScript:

```js
const x = 1

(a) => {}
```

Without a semicolon after `const x = 1`, the parser interprets `1(a)` as a call expression on the numeric literal `1`, then hits `=>` and fails:

```
Unexpected token =>
```

The issue appears to originate from the FunctionDeclaration and ClassDeclaration removal paths in handleFixes, which call fixer.removeRange(parent.range) directly and do not appear to perform the ASI-safety checks used for VariableDeclarator removals

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgeCA9IDFcbmZ1bmN0aW9uIHVudXNlZCgpIHt9XG4oYSkgPT4ge31cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Also affects `ClassDeclaration`:

```js
const x = 1
class Unused {}
(a) => {}
```

Applying the `removeVar` suggestion produces the same invalid output and parser error.

For comparison, in an ASI-sensitive variable-declaration scenario, ESLint still reports the unused variable but suppresses the removal suggestion. This suggests that variable removals already have protection against unsafe declaration removal, while function and class removals do not.

To reproduce the issue, apply the removeVar suggestion for unused.

**Comment by DMartens** [comment]
Thank you for catching this, feel free to open a PR.
But this is not limited to functions and classes.
For example removing "y" in the following example also would lead to a syntax error:
```javascript
const x = 1,
      y = () => {}
() => {};
```

**Comment by Kuldeep2822k** [comment]
>  For example removing "y" in the following example also would lead to a syntax error:
> 

oh i totally misssed that will extend safety check
